pkg/apisix/cluster.go

```diff
@@ -63,6 +77,7 @@ type ClusterOptions struct {
 type cluster struct {
 	name         string
 	baseURL      string
+	baseURLHost  string
```

Comment:
If host is required, we can change the `baseURL` to `pathPrefix`, currently, `baseURLHost` is partial of `baseURL`.

Reply:
Split baseURL to pathPrefix and baseURLHost is not enough. Also need scheme. So how to change this config ?

Reply:
Or you can just save the `url.URL` object.

Reply:
Replace baseURL with *url.URL object may be make more changes. Can we do that in this pr ? Or just use baseURLHost.

Reply:
OK, agree with you :), we can change them in subsequent PRs.